src/utils/SpokeUtils.ts

```diff
@@ -0,0 +1,142 @@
+import { SpokePool } from "../typechain";
+
+/**
+ * Find the block range that contains the deposit ID. This is a binary search that searches for the block range
+ * that contains the deposit ID.
+ * @param targetDepositId The target deposit ID to search for.
+ * @param initLow The initial lower bound of the block range to search.
+ * @param initHigh The initial upper bound of the block range to search.
+ * @param maxSearches The maximum number of searches to perform. This is used to prevent infinite loops.
+ * @returns The block range that contains the deposit ID.
+ * @note  // We want to find the block range that satisfies these conditions:
+ *        // - the low block has deposit count <= targetDepositId
+ *        // - the high block has a deposit count > targetDepositId.
+ *        // This way the caller can search for a FundsDeposited event between [low, high] that will always
+ *        // contain the event emitted when deposit ID was incremented to targetDepositId + 1. This is the same transaction
+ *        // where the deposit with deposit ID = targetDepositId was created.
+ */
+export async function getBlockRangeForDepositId(
+  targetDepositId: number,
+  initLow: number,
+  initHigh: number,
+  maxSearches: number,
+  spokePool: SpokePool,
```

Comment:
Does this need to be a `SpokePool` type, or could we live with a regular ethers `Contract` definition?

Reply:
I personally prefer to be as explicit with the typecasting as possible. Do you think it negatively affects runtime? Happy to switch it back

Reply:
I guess I don't really see any benefits from it, vs. a standard Ethers `Contract` type. A standard Ethers `Contract` seems preferable to me because it's more portable and imposes the least on the caller. I'm pretty sure you could swap the type back to `Contract` and it'd work without any further modifications at all.

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/383/commits/08e9de5d6fedb88a3384d25720ce712b2be04fd8